### PR TITLE
Allow `SimpleAccount` to transfer native token with `executeBatch`

### LINKED ISSUE #275: Add "value" parameter to `SimpleAccount.executeBatch`
Why doesn't the `SimpleAccount.executeBatch` method take a `value` parameter like normal `execute`? For example, encode multiple transfer in `executeBatch` is better then execute a multicall.

[execute](https://github.com/eth-infinitism/account-abstraction/blob/75f02457e71bcb4a63e5347589b75fa4da5c9964/contracts/samples/SimpleAccount.sol#LL67C1-L68C1):
`function execute(address dest, uint256 value, bytes calldata func)`

[executeBatch](https://github.com/eth-infinitism/account-abstraction/blob/75f02457e71bcb4a63e5347589b75fa4da5c9964/contracts/samples/SimpleAccount.sol#L75):
`function executeBatch(address[] calldata dest, bytes[] calldata func)`

revised `executeBatch` with value:
`function executeBatch(address[] calldata dest, uint256[] value, bytes[] calldata func)`

it was originally introduced for `DepositPaymaster`, I think it can have broader use after the modification.

### PR DESCRIPTION
**Pull request by kaiix**
Resolves #275

### CODE REVIEW

File: contracts/samples/SimpleAccount.sol [hunk 1]
```diff
@@ -62,11 +62,11 @@ contract SimpleAccount is BaseAccount, TokenCallbackHandler, UUPSUpgradeable, In
     /**
      * execute a sequence of transactions
      */
-    function executeBatch(address[] calldata dest, bytes[] calldata func) external {
+    function executeBatch(address[] calldata dest, uint256[] calldata value, bytes[] calldata func) external {
         _requireFromEntryPointOrOwner();
         require(dest.length == func.length, "wrong array lengths");
```
Comment (drortirosh) [inline contracts/samples/SimpleAccount.sol]: should also verify array length.

File: contracts/samples/SimpleAccount.sol [hunk 2]
```diff
@@ -62,11 +62,11 @@ contract SimpleAccount is BaseAccount, TokenCallbackHandler, UUPSUpgradeable, In
...
         for (uint256 i = 0; i < dest.length; i++) {
-            _call(dest[i], 0, func[i]);
+            _call(dest[i], value[i], func[i]);
```
Comment (drortirosh) [inline contracts/samples/SimpleAccount.sol]: this makes the trivial (no-value) case more expensive, as it requires yet another array.
maybe this array-of-values be optional. putting a zero-length array to mean zero value.

**Comment by drortirosh** [comment]
please run `yarn gas-checker`

**Comment by kaiix** [comment]
> please run `yarn gas-checker`

done, generated a new gas report by running `yarn gas-calc`

**Review by drortirosh — APPROVED** [review]

**Review by drortirosh — COMMENTED** [review]
please rebase your PR before it can be merged. while it can be merged, gas-calcs tend to drift with seemingly unrelated commits.
sorry for the trouble...

**Comment by kaiix** [comment]
> please rebase your PR before it can be merged. while it can be merged, gas-calcs tend to drift with seemingly unrelated commits. sorry for the trouble...

rebased and re-runned the gas-calcs (nothing changes)

**Comment by hongqn** [comment]
@drortirosh Is this pullreq ready to merge?

**Review by drortirosh — APPROVED** [review]

**Pull request merged**